src/main/java/com/google/devtools/build/lib/skyframe/IgnoredPackagePrefixesFunction.java

```diff
@@ -91,10 +93,19 @@ public SkyValue compute(SkyKey key, Environment env)
         }
       } else {
         // Make sure the repository is fetched.
-        RepositoryDirectoryValue repositoryValue =
-            (RepositoryDirectoryValue) env.getValue(RepositoryDirectoryValue.key(repositoryName));
-        if (repositoryValue == null) {
-          return null;
+        RepositoryDirectoryValue repositoryValue;
+        try {
+          repositoryValue =
+              (RepositoryDirectoryValue) env
+                  .getValueOrThrow(RepositoryDirectoryValue.key(repositoryName),
+                      RepositoryFunctionException.class, IOException.class);
```

Comment:
I don't think this case can happen? When we evaluate a SKyKey with `getValueOrThrow`, the thrown exception is `ErrorInfo.getCause()` [1], which is basically unwrapping the `SkyFunctionException` to get the root cause exception [2]. So what you're looking for here should be one of the wrapped exceptions that could be thrown when evaluating the skykey with `RepositoryFunction` [3].

[1] https://cs.opensource.google/bazel/bazel/+/master:src/main/java/com/google/devtools/build/skyframe/SkyFunctionEnvironment.java;l=603;drc=b26a1a5be07be1f2200daed8bba94ac955518c6a 
[2] https://cs.opensource.google/bazel/bazel/+/master:src/main/java/com/google/devtools/build/skyframe/ErrorInfo.java;l=39;drc=a5a4f47295ccb6cd81cf8732772c849b1a8b9d58
[3] https://cs.opensource.google/bazel/bazel/+/master:src/main/java/com/google/devtools/build/lib/rules/repository/RepositoryFunction.java;l=98-119;drc=0617f2c7b3f442a00f7359d216b8fab14d2dd762

Reply:
Hmm, I think you're right, I'll update it on the internal change (since I've already imported it).